src/engine/contactwriter.cpp

```diff
@@ -3071,6 +3071,10 @@ static bool removeContactDetails(QContact *contact, const QList<StringPair> &rem
     return true;
 }
 
+// Despite the similarity in name between this function and modifyContactDetail,
+// this function is used when saving modifications to an aggregate contact.
+// Those updates get downpromoted to the appropriate constituent contact in this
+// function.
 static bool modifyContactDetails(QContact *contact, const QList<QPair<StringPair, QContactDetail> > &updates)
```

Comment:
Right, but no amount of comment is as good as an improved function name... perhaps "applyDetailUpdates"?